### PR TITLE
Adds UDS and a no-stdout mode with builtin throughput

### PR DESCRIPTION
**Pull request by scottopell**
Are you interested in any of these changes, I can break up into smaller commits no problem.

- Accept incoming data via UDS
- Don't output to stdout by default and instead use very basic throughput printer
- noop rust mode
- rustfmt changes
- rust `replace` -> `replacen`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -51,6 +50,13 @@ pub extern "C" fn transform(input: *const libc::c_char) -> *const libc::c_char {
     return c_str.into_raw();
 }
 
+#[no_mangle]
+pub extern "C" fn noop(input: *const libc::c_char) -> *const libc::c_char {
```
Comment (gh123man) [inline src/lib.rs:54]: I think this is the same as `passthrough` so you can remove that function.

Comment (gh123man) [reply src/lib.rs:54]: same with line 187 on `main.go`

**Review by gh123man — APPROVED** [review]
just a small comment on passthrough vs noop.

File: src/lib.rs (hunk 1 above)
Comment (scottopell) [reply src/lib.rs:54]: 👍

**Review by gh123man — APPROVED** [review]

**Pull request merged**